src/source/Ice/SocketConnection.c

```diff
@@ -121,16 +121,21 @@ STATUS socketConnectionInitSecureConnection(PSocketConnection pSocketConnection,
     }
 
     SSL_set_mode(pSocketConnection->pSsl, SSL_MODE_AUTO_RETRY);
-    CHK((pSocketConnection->pWriteBio = BIO_new_fd(pSocketConnection->localSocket, BIO_NOCLOSE)) != NULL, STATUS_SSL_CTX_CREATION_FAILED);
     CHK((pSocketConnection->pReadBio = BIO_new(BIO_s_mem())) != NULL, STATUS_SSL_CTX_CREATION_FAILED);
+    CHK((pSocketConnection->pWriteBio = BIO_new(BIO_s_mem())) != NULL, STATUS_SSL_CTX_CREATION_FAILED);
```

Comment:
these is related actually. Previously we use a fd Bio which writes to the socket directly. But since we need to also pass the NO_SIGNAL flag to send(), and openssl doenst expose that through SSL_write, we have to switch to use a mem bio instead and do the send ourselves.

Reply:
I see, makes sense now.